languages/csharp/exercises/concept/chars/.meta/Example.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using System.Text;
+
+public static class CharUtils
+{
+    public static bool DetectInvalidCharComination(string str)
+    {
+        const char DASH = '-';
+        const char GREATER_THAN = '>';
+        for (int i = 1; i < str.Length - 1; i++)
+        {
+            if (str[i] == DASH && str[i + 1] == GREATER_THAN && str[i - 1] != DASH)
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public static string ToUpper(string str)
+    {
+        var sb = new StringBuilder();
+        foreach (char ch in str)
+        {
+            sb.Append(Char.ToUpper(ch));
+        }
+
+        return sb.ToString();
+    }
+
+    public static string CleanIdentifier(string str)
```

Comment:
There is so much in this single method that I feel like this could perhaps be the single (or with one other) method the student has to implement. We could progressively add constraints to build up a fully-fleshed solution.

Reply:
> 1. I agree that some sort of staged approach might help the student.

👍 

> 2. We can include conversion of kebab-case to camelCase here and some variation of validation of a C pointer.

Sure, as long as we can keep the exercise relatively easy to implement. We don't want too much variation in possible implementations, as otherwise there will be (too) many representations.

> 3. I would like to keep 2 ordering/comparison exercises to contrast the way we normally do it (using operators) with how we should do it (using CompareTo) method. I need to investigate the role of CompareTo.

Interesting. The `design.md` states that "Advanced unicode issues such as surrogates, text normalization, combining characters, sort order" is out of scope. I had originally envisioned that `CompareTo` would be part of a stand-alone exercise that deals with `IComparable` and comparing objects. I'm slightly worried that also including it here might lead to the exercise to do a bit too much, as it will then introduce chars, StringBuilder and comparison. That said, we might get away with just mentioning `CompareTo` as an alternative, but I'm not completely sure here.

> 4. I am concerned that a large exercise will create too many degrees of freedom for the student so as to require for too many variations of "model answer" or whatever we are calling the thing that the presenter compares.

100% agree. Let's try to keep the exercise as minimal as possible. I'll let you work on a new iteration and then we can see what it looks like.

Reply:
I have removed `CompareTo`.  I thought it did some magic (although that did not make sense) that simple comparison operators did not.  I also liked it as an instance member of `char` but I don't think that is crucial.

Reply:
I think that's a good call. We also have `Char.IsWhiteSpace` and `Char.IsControl` for the instance member.

Reply:
No.  only `Equals` and `ToString`.  I'll assume you're off in ruby-land coding your site.

Reply:
Oh sorry, I misread! I'll also try and think of some instance member, although there is indeed not much to work with there. `ToString` might be our best bet here, but with the use of string interpolation, even that doesn't make much sense in most cases. Maybe this is one of the cases where we just mention that there are instance members, but don't have the tests verify using them.

Reply:
> just mention that there are instance members

I'll do that and we can revisit if necessary.